src/core/util/StringBuilder.ts

```diff
@@ -1,10 +1,15 @@
+import StringEncoding from './StringEncoding';
+
 export default class StringBuilder {
 
-  public constructor(private value: string = '') {}
+  public constructor(private value: string = '', private format = null) {}
 
   public append(s: string | number): StringBuilder {
     if (typeof s === 'string') {
       this.value += s.toString();
+    } else if (this.format) {
```

Comment:
`StringBuilder` in Java does not have a format parameter, so could you explain me why we need and why this should exist in here? Also, is there some alternative solution for that? We have a concern here that all the code should be as close as possible to the Java version so it can make it easier for future contributors to engage and understand.

Reply:
The reason:
as I see, in Java the statement (char) x will return symbol in ASCII (not UTF-8 as in JS charCodeAt) encoding and that's why some tests not passed (if result contain symbols with code > 127)
Maybe we can use UTF-8 but I'm not sure

Reply:
Couldn't we fix the string encoding problems before sending it to the `StringBuilder` class? It would be nice to keep `StringBuilder` as pristine as possible to avoid problems with unexpected behaviors in the future.